ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/tests/PartyManagement.scala

```diff
@@ -183,4 +183,62 @@ final class PartyManagement(session: LedgerSession) extends LedgerTestSuite(sess
           s"The allocated party IDs $allocatedPartyIds are not a subset of $knownPartyIds.")
       }
   }
+
+  test(
+    "PMListKnowPartiesIsLocal",
+    "Listing parties on multi-node ledgers should reflect whether parties are local",
+    allocate(NoParties, NoParties),
+  ) {
+    case Participants(Participant(alpha), Participant(beta)) =>
+      for {
+        alphaLocalParty <- alpha.allocateParty(
+          partyIdHint = Some("PMListKnowPartiesIsLocal_" + Random.alphanumeric.take(10).mkString),
+          displayName = Some("Alice"),
+        )
+        betaLocalParty <- beta.allocateParty(
+          partyIdHint = Some("PMListKnowPartiesIsLocal_" + Random.alphanumeric.take(10).mkString),
+          displayName = Some("Bob"),
+        )
+        // only wait for parties on ledgers configured to know about each other's parties
+        _ <- alpha.waitForParties(Set(alpha, beta), Set(alphaLocalParty, betaLocalParty))
+        _ <- beta.waitForParties(Set(alpha, beta), Set(alphaLocalParty, betaLocalParty))
+        partyDetailsKnowToAlpha <- alpha.listKnownPartyDetails()
+        partyDetailsKnowToBeta <- beta.listKnownPartyDetails()
+      } yield {
+        def lookUp(
+            party: binding.Primitive.Party,
+            details: Seq[PartyDetails]): Option[(String, Boolean)] =
+          details.collectFirst {
+            case PartyDetails(p, displayName, isLocal)
+                if p == binding.Primitive.Party.unwrap(party) =>
+              (displayName, isLocal)
+          }
```

Comment:
Rather than having this bespoke method here, use `getParties`, passing `alphaLocalParty` and `betaLocalParty` in. This prevents the tests from possibly interfering with others that may use `Alice` and `Bob` as party hints.

Reply:
Done thanks for the pointer to `getParties` (should have noticed it myself while looking through the helpers).